localstack/services/lambda_/invocation/lambda_service.py

```diff
@@ -291,6 +292,16 @@ def invoke(
         # empty payloads have to work as well
         if payload is None:
             payload = b"{}"
+        else:
+            # detect invalid payloads early before creating an execution environment
+            try:
+                to_str(payload)
```

Comment:
I did not test whether AWS also detects this parsing error early but guess that makes sense. I guess double `to_str` conversion should not be too problematic given that there are payload limits.

Limits: https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html

```
6 MB each for request and response (synchronous)

20 MB for each [streamed response](https://docs.aws.amazon.com/lambda/latest/dg/configuration-response-streaming.html) (Synchronous. The payload size for streamed responses can be increased from default values. Contact AWS Support to inquire further.)

256 KB (asynchronous)
```